onebox.sh

```diff
@@ -0,0 +1,311 @@
+        TEMPLATE_TYPE=devstack
+    fi
+
+    # Allow for synonyms
+    if [[ $BRANCH_VERSIONS == production ]] || [[ $BRANCH_VERSIONS == prod ]] || [[ $BRANCH_VERSIONS == master ]] || [[ $BRANCH_VERSIONS == release ]] ; then
+        BRANCH_VERSIONS=stable
+    elif [[ $BRANCH_VERSIONS == development ]] || [[ $BRANCH_VERSIONS == dev ]] || [[ $BRANCH_VERSIONS == beta ]] || [[ $BRANCH_VERSIONS == pre ]] || [[ $BRANCH_VERSIONS == int ]] ; then
+        BRANCH_VERSIONS=edge
+    elif [[ $BRANCH_VERSIONS == upstream ]] || [[ $BRANCH_VERSIONS == up ]] || [[ $BRANCH_VERSIONS == ed ]] ; then
+        BRANCH_VERSIONS=edx
+    fi
+}
+
+test_args()
+{
+    if [[ $TEMPLATE_TYPE != fullstack ]] && [[ $TEMPLATE_TYPE != devstack ]] ; then
+        set +x
+        echo -e "\033[1;36m"
+        echo -e "\n TEMPLATE_TYPE is set to $TEMPLATE_TYPE"
+        echo -e " but should be fullstack or devstack."
+        echo -e " Use the -r param argument.\n"
+        echo -e '\033[0m'
+        exit 1
+    fi
+
+    if [[ $BRANCH_VERSIONS != stable ]] && [[ $BRANCH_VERSIONS != edge ]] && [[ $BRANCH_VERSIONS != edx ]] ; then
+        set +x
+        echo -e "\033[1;36m"
+        echo -e "\n BRANCH_VERSIONS is set to $BRANCH_VERSIONS"
+        echo -e " but should be stable OR edge OR edx .\n"
+        echo -e " Use the -b param argument.\n"
+        echo -e '\033[0m'
+        exit 1
+    fi
+}
+
+##########################
+# Helpers
+##########################
+get_branch()
+{
+    if [[ $BRANCH_VERSIONS == stable ]] ; then
+        echo "oxa/master.fic"
+    elif [[ $BRANCH_VERSIONS == edge ]] || [[ -n $1 ]] ; then
+        if [[ -n $2 ]] ; then
+            # Legacy switch
+            echo "oxa/devfic"
+        else
+            echo "oxa/dev.fic"
+        fi
+    elif [[ $BRANCH_VERSIONS == edx ]] ; then
+        echo "$EDX_BRANCH"
+    else
+        test_args
+    fi
+}
+#todo: switch to get_branch() after platform change is merged
+get_plat_branch()
+{
+    if [[ $BRANCH_VERSIONS == stable ]] ; then
+        echo "oxa/master.fic"
+    elif [[ $BRANCH_VERSIONS == edge ]] ; then
+        echo "oxa/df_noConfig"
+    elif [[ $BRANCH_VERSIONS == edx ]] ; then
+        echo "$EDX_BRANCH"
+    else
+        test_args
+    fi
+}
+
+get_current_branch()
+{
+    prefix='* '
+
+    # Current branch is prefixed with an asterisk. Remove it.
+    branchInfo=`git branch | grep "$prefix" | sed "s/$prefix//g"`
+
+    # Ensure branch information is useful.
+    if [[ -z "$branchInfo" ]] || [[ $branchInfo == *"no branch"* ]] || [[ $branchInfo == *"detached"* ]] ; then
+        #todo: uncomment before merge
+        #branchInfo=`get_branch useMicrosoftRepo oldDevStyle`
+        branchInfo="oxa/df_noConfig"
+    fi
+
+    echo "$branchInfo"
+}
+
+harden()
+{
+    # Is the current password insecure?
+    if [[ -z $1 ]] || [[ $1 == $DEFAULT_STRING ]] ; then
+        if [[ -n $DEFAULT_PASSWORD ]] ; then
```

Comment:
@kabirkhan (click "show outdated" to see this on the webpage). supporting empty passwords doesn't work for mongo and local admin user. i'll revert support for that in my next commit